### PR TITLE
Use starknet_rs BlockId implementation

### PR DESCRIPTION
**Pull request by marioiordanov**
## Development related changes

Remove BlockId implementation from types crate and use starknet_rs instead

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Documented the changes
- [ ] Linked the issues which this PR resolves
- [ ] Updated the tests
- [ ] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet-server/src/api/json_rpc/models.rs [hunk 1]
```diff
@@ -457,9 +452,9 @@ mod tests {
             r#"{"block_id": {"block_hash": "0x004134134134134134134134134134134134134134134134134134134134134134"}}"#,
         );
 
-        // Block hash hex doesnt start with 0x
+        // Changing to starknet_rs BlockId, now those cases can be handled - hex to be non prefixed
         assert_block_id_block_hash_correctness(
-            false,
+            true,
             "0x01",
             r#"{"block_id": {"block_hash": "01"}}"#,
```
Comment (FabijanC) [inline crates/starknet-server/src/api/json_rpc/models.rs:459]: But should this really be correct? What is this test testing? The correctness of `FieldElement` implementation or something in our code?

Comment (marioiordanov) [reply crates/starknet-server/src/api/json_rpc/models.rs:459]: In our implementation hex strings which were not prefixed with 0x were not possible. Starknet-rs accept such hex values

Comment (FabijanC) [reply crates/starknet-server/src/api/json_rpc/models.rs:459]: Ok, are we then diverging from the specification in some cases? By not being strict enough?

**Comment by marioiordanov** [comment]
still being able to deserialize non hex prefixed strings as hex ones. For example 0a is deserialized as 0x0a

**Pull request closed without merging**